src/hotspot/share/gc/shenandoah/shenandoahEvacTracker.cpp

```diff
@@ -89,24 +99,27 @@ void ShenandoahEvacuationTracker::print_evacuations_on(outputStream* st,
   mutators->print_on(st);
   st->cr();
 
-  AgeTable young_region_ages(false), old_region_ages(false);
   ShenandoahHeap* heap = ShenandoahHeap::heap();
-  for (uint i = 0; i < heap->num_regions(); ++i) {
-    ShenandoahHeapRegion* r = heap->get_region(i);
-    if (r->age() > 0 && r->age() < AgeTable::table_size) {
-      if (r->is_young()) {
-        young_region_ages.add(r->age(), r->get_live_data_words());
-      } else {
-        assert(r->is_old(), "Sanity");
-        old_region_ages.add(r->age(), r->get_live_data_words());
+  if (_generational) {
+    AgeTable young_region_ages(false), old_region_ages(false);
+    for (uint i = 0; i < heap->num_regions(); ++i) {
+      ShenandoahHeapRegion* r = heap->get_region(i);
+      if (r->age() > 0 && r->age() < AgeTable::table_size) {
+        if (r->is_young()) {
+          young_region_ages.add(r->age(), r->get_live_data_words());
+        } else {
+          assert(r->is_old(), "Sanity");
+          old_region_ages.add(r->age(), r->get_live_data_words());
```

Comment:
We currently don't do anything with this. Perhaps the idea is to use it in some way in the future?

Reply:
Leaving as is for now for informational purposes; we can deal with this suitably in the future if/when we use it in some way. We might also decide to elide it until such time.